internal/impl/shell.go

```diff
@@ -87,28 +87,34 @@ func NewDevboxShell(nixpkgsCommitHash string, opts ...ShellOption) (*DevboxShell
 }
 
 // shellPath returns the path to a shell binary, or error if none found.
-func shellPath(nixpkgsCommitHash string) (path string, err error) {
+func shellPath(nixpkgsCommitHash string, pure bool) (path string, err error) {
 	defer func() {
 		if err != nil {
 			path = filepath.Clean(path)
 		}
 	}()
 
-	// First, check the SHELL environment variable.
-	path = os.Getenv(envir.Shell)
-	if path != "" {
-		debug.Log("Using SHELL env var for shell binary path: %s\n", path)
-		return path, nil
+	if !pure {
+		// First, check the SHELL environment variable.
+		path = os.Getenv(envir.Shell)
+		if path != "" {
+			debug.Log("Using SHELL env var for shell binary path: %s\n", path)
+			return path, nil
+		}
 	}
 
 	// Second, fallback to using the bash that nix uses by default.
 
 	var bashNixStorePath string // of the form /nix/store/{hash}-bash-{version}/
 
-	cmd := exec.Command(
-		"nix", "eval", "--raw",
-		fmt.Sprintf("%s#bash", nix.FlakeNixpkgs(nixpkgsCommitHash)),
-	)
+	// construct exec args, add --pure-eval if pure shell
+	execArgs := []string{"eval", "--raw"}
+	if pure {
+		execArgs = append(execArgs, "--pure-eval")
```

Comment:
🤔 I'm not super familiar and didn't find good nix docs about this.

1. Why do we need it? Could you point to some good docs you may have found about it?
2. Should we also add it to `print-dev-env` call inside `nix.computeEnv`? Hmm but that would complicate the printDevEnv cache that we have.

Reply:
https://docs.haskellstack.org/en/v2.9.1/nix_integration/#pure-and-impure-nix-shells

This is the doc I found. Which aligns with our approach for a pure shell.

Reply:
This is just trying to find the shell to use right? I don't think this needs `pure-eval` (it's unrelated to the shell created for the user)